parachain/pallets/control/src/lib.rs

```diff
@@ -170,6 +208,11 @@ pub mod pallet {
 				location
 			);
 
+			Self::reserve_deposit(
+				para_id,
```

Comment:
Specifically for `CreateAgent`, we should be transferring the fee from the sovereign account of `origin_location` rather than the parachain that `origin_location` lives on.

Often the governance of parachains won't have control over which sub-accounts make XCM calls into BridgeHub, and so for example, malicious EVM contracts on Moonbeam could can spam `create_agent` calls and they won't have to pay for it, as the fees will be deducted from the parachain's sovereign account.

So you'll need some kind of converter that converts `origin_location` into a signed sovereign account.